packages/arb-token-bridge-ui/src/pages/api/deposits.ts

```diff
@@ -0,0 +1,112 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import { gql } from '@apollo/client'
+import {
+  FetchDepositsFromSubgraphResult,
+  getL1SubgraphClient
+} from 'token-bridge-sdk'
+
+// Extending the standard NextJs request with Deposit-params
+type NextApiRequestWithDepositParams = NextApiRequest & {
+  body: {
+    l2ChainId: number
+    address?: string
+    search?: string
+    page?: number
+    pageSize?: number
+    fromBlock?: number
+    toBlock?: number
```

Comment:
Thinking maybe these two should always be required to not put too much load on the subgraph? Possibly even validating that it's below a certain threshold? :thinking: